### PR TITLE
Added leaves table and Active record relationships

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -27,5 +46,9 @@
     t.string   "refresh_token"
     t.string   "email"
     t.string   "avatar"
+    t.boolean  "is_admin"
```
Comment (revathskumar) [inline db/schema.rb]: We don't need `is_admin` and `calendar` schema updates in this PR. will cause conflict when other PR's merged to master.

File: db/schema.rb [hunk 2]
```diff
@@ -23,18 +23,18 @@
     t.integer  "user_id"
   end
 
-  create_table "users", force: :cascade do |t|
```
Comment (revathskumar) [inline db/schema.rb]: This PR shouldn't remove users table from schema. It should be kept as unchanged.

File: db/migrate/20160701095455_create_leaves.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateLeaves < ActiveRecord::Migration[5.0]
+  def change
+    create_table :leaves do |t|
+      t.string :summary
+      t.datetime :start
+      t.datetime :end
+      t.references :user, foreign_key: true
+      t.references :calendar, foreign_key: true
```
Comment (revathskumar) [inline db/migrate/20160701095455_create_leaves.rb]: @HarshwardhanSingh I think we need to keep a different table to map leave to calendar since we ill be updating multiple calendars for the same leave.

Comment (HarshwardhanSingh) [inline db/migrate/20160701095455_create_leaves.rb]: @revathskumar I think this leaves table can be used to achieve the same.

Comment (revathskumar) [inline db/migrate/20160701095455_create_leaves.rb]: Can you explain you are planning to do with leaves table?

Comment (HarshwardhanSingh) [inline db/migrate/20160701095455_create_leaves.rb]: Each time when a user wants to mark leave we can display his calendars via checkbox and can add multiple records depending on the no of calendars to mark.

File: db/migrate/20160705071038_create_leavecalendars.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateLeavecalendars < ActiveRecord::Migration[5.0]
+  def change
+    create_table :leaves_calendars do |t|
```
Comment (revathskumar) [inline db/migrate/20160705071038_create_leavecalendars.rb]: I think the association name should be in alphabetical order. Not sure if this is still applicable. 
[source](http://stackoverflow.com/questions/2780798/has-and-belongs-to-many-vs-has-many-through#comment2816466_2781049) 
So instead `leaves_calendar` it should be `calendars_leaves` will be the proper name.

File: db/schema.rb [hunk 5]
```diff
@@ -23,6 +23,34 @@
     t.integer  "user_id"
   end
 
+  create_table "invites", force: :cascade do |t|
```
Comment (revathskumar) [inline db/schema.rb]: Invites doesn't belong in this PR.

File: db/migrate/20160701095455_create_leaves.rb (hunk 3 above)
Comment (revathskumar) [inline db/migrate/20160701095455_create_leaves.rb]: We don't need calendar reference in this table

File: app/models/leavecalendar.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class Leavecalendar < ApplicationRecord
```
Comment (revathskumar) [inline app/models/leavecalendar.rb]: We don't need to create this model. AFAIK rails will take care of this.

File: spec/models/leavecalendar_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (revathskumar) [inline spec/models/leavecalendar_spec.rb]: This file is not required when you remove the LeaveCalendar model.

Comment (HarshwardhanSingh) [inline spec/models/leavecalendar_spec.rb]: @revathskumar So should I just delete spec/models/leavecalendar_spec.rb file
